src/worker.js

```diff
@@ -367,8 +367,37 @@ const aioli = {
 				}
 			},
 			// Setup print functions to store stdout/stderr output
-			print: text => tool.stdout += `${text}\n`,
-			printErr: aioli.config.printInterleaved ? text => tool.stdout += `${text}\n` : text => tool.stderr += `${text}\n`
+			print: text => {
+				tool.stdout += `${text}\n`;
+				if(this.config.printStream)
+					postMessage({
+						type: "biowasm",
+						value: {
+							stdout: text,
+						},
+					});
+			},
+			printErr: aioli.config.printInterleaved ?
+				text => {
+					tool.stdout += `${text}\n`;
+					if(this.config.printStream)
+						postMessage({
+							type: "biowasm",
+							value: {
+								stdout: text,
+							},
+						});
+				} :
+				text => { 
+					tool.stderr += `${text}\n`;
+					if(this.config.printStream)
+					postMessage({
+						type: "biowasm",
+						value: {
+							stderr: text,
+						},
+					});
+				}
```

Comment:
We could make this part of the code a little smaller:

```suggestion
			printErr: text => {
				const destination = aioli.config.printInterleaved ? "stdout" : "stderr";
				tool[destination] += `${text}\n`;
				if(aioli.config.printStream)
					postMessage({
						type: "biowasm",
						value: {
							[destination]: text,
						},
					});
				}
			}
```

I haven't actually tested this but it should work 🙂